pkgs/by-name/sk/skyemu/package.nix

```
@@ -0,0 +1,99 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  ninja,
+  pkg-config,
+  curl,
+  openssl,
+  SDL2,
+  alsa-lib,
+  libGL,
+  libGLU,
+  libX11,
+  libXi,
+  libXcursor,
+  lua,
+  makeDesktopItem,
+  copyDesktopItems,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "skyemu";
+  version = "3-unstable-2025-02-23";
+
+  src = fetchFromGitHub {
+    owner = "skylersaleh";
+    repo = "SkyEmu";
+    rev = "f8573db83d15791b0cd94c29ceb46bf683963ff0";
+    hash = "sha256-LI4zBKjB48zYYYVZePzefRFrw/zvg17jzh5ZOPHAWok=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+    pkg-config
+    copyDesktopItems
+  ];
+
+  buildInputs =
+    [
+      curl
+      openssl
+      SDL2
+      alsa-lib
+      libGL
+      libGLU
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      libX11
+      libXi
+      libXcursor
+      lua
+    ];
+
+  cmakeFlags = [
+    "-DUSE_SYSTEM_CURL=ON"
+    "-DUSE_SYSTEM_OPENSSL=ON"
+    "-DUSE_SYSTEM_SDL2=ON"
+    "-DENABLE_RETRO_ACHIEVEMENTS=ON"
+  ];
+
+  postPatch = ''
+    substituteInPlace CMakeLists.txt \
+      --replace 'set(CMAKE_OSX_ARCHITECTURES' '#set(CMAKE_OSX_ARCHITECTURES'
+  '';
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "skyemu";
+      exec = "SkyEmu";
+      icon = "skyemu";
+      comment = "GameBoy, GameBoy Color, GameBoy Advance, and DS emulator";
+      desktopName = "SkyEmu";
+      categories = [
+        "Game"
+        "Emulator"
+      ];
+    })
+  ];
+
+  postInstall = ''
+    mkdir -p $out/share/pixmaps
+
+    if [ -d src/resources/icons ]; then
+      if [ -f src/resources/icons/AppIcon.png ]; then
+        cp src/resources/icons/AppIcon.png $out/share/pixmaps/skyemu.png
+      fi
+    fi
+  '';
```

```suggestion
  postInstall = ''
    install -Dm644 src/resources/icons/AppIcon.png $out/share/pixmaps/skyemu.png
  '';
```

Let it fail if icon isn't installed